_posts/2017-05-15-rust-at-two-years.md

```diff
@@ -0,0 +1,356 @@
+nightly users. It provides type information on hover, error messages as you
+type, and different kinds of code navigation. It even provides refactoring and
+formatting as unstable features! It works with projects as large as Cargo.
+We're excited to watch the RLS continue to grow and hope to see it make its way
+to stable Rust later this year.
+
+[Rust Language Server]: https://github.com/rust-lang-nursery/rls
+[available from rustup]: https://github.com/rust-lang-nursery/rls#setup
+
+*Thanks to Jonathan Turner for this RLS summary!*
+
+### [Rust should have 1.0-level crates for essential tasks](https://github.com/rust-lang/rust-roadmap/issues/11), and [Rust should provide easy access to high quality crates](https://github.com/rust-lang/rust-roadmap/issues/9)
+
+The [recent post on the Libz Blitz] details the Library Team's initiative to
+increase the quality of crates for common tasks; that post is excellent so I
+won't repeat it here. I will note that many of the issues that the Libs Team
+is going to create will be great starter issues. For the blitz to be the best
+it can be, the Libs Team is going to need help from the community-- that means
+YOU! :) They're willing to mentor people interested in contributing.
+
+In order to make awesome crates easier to find for particular purposes,
+crates.io now has [categories] for crate authors to better indicate the use
+case of their crate. Crates can also now have CI badges, and
+[more improvements to crates.io's interface] are coming that will help you
+choose the crates that fit your needs.
+
+[recent post on the Libz Blitz]: https://blog.rust-lang.org/2017/05/05/libz-blitz.html
+[categories]: https://crates.io/categories
+[more improvements to crates.io's interface]: https://github.com/rust-lang/rust/issues/41616
+
+### [Rust should be well-equipped for writing robust, high-scale servers](https://github.com/rust-lang/rust-roadmap/issues/10)
+
+TODO: [Futures] and [tokio] enable zero-cost asynchronous programming
+
+[Futures]: https://crates.io/crates/futures
+[tokio]: https://tokio.rs/
+
+### [Rust should integrate easily into large build systems](https://github.com/rust-lang/rust-roadmap/issues/12)
+
+This initiative is mostly in the ideas stage; there's lots of great discussion
+on the tracking issue that has resulted in a few Cargo issues:
+
+* [Support creating a build plan](https://github.com/rust-lang/cargo/issues/3815)
+* [Support declaring external dependencies](https://github.com/rust-lang/cargo/issues/3816)
+
+Keep an eye out for more improvement in this area soon.
+
+### [Rust's community should provide mentoring at all levels](https://github.com/rust-lang/rust-roadmap/issues/13)
+
+There were [RustBridge] events before RustFest Berlin and Rust Belt Rust, and
+there's another planned for the day before RustConf coming up!
+
+[RustBridge]: https://github.com/rust-community/rustbridge
+
+There are many people who have started getting involved with various
```

Comment:
Worth talking about:

- The [introduction](https://internals.rust-lang.org/t/language-team-shepherds/4595) of language team "shepherds", which is partly about mentoring a set of folks around the lang team (and has been somewhat successful at that).
- The Libz Blitz is set up specifically to provide a lot of mentoring opportunities. We did something similar with [error message improvements](https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html)
- We've been working to grow both the number and size of subteams, to create more opportunities for people to step into leadership roles.
- Have Outreachy and GSOC interns this summer.

We should emphasize the "all levels" part of the roadmap item: it's about onboarding first-time contributors but also adding folks all the way up at the core team level.